core/trino-main/src/main/java/io/trino/sql/planner/planprinter/JsonRenderer.java

```diff
@@ -54,6 +83,88 @@ private JsonRenderedNode renderJson(PlanRepresentation plan, NodeRepresentation
                         .collect(toImmutableList()));
     }
 
+    public Map<Integer, List<DistributedPlanRepresentation>> render(PlanPrinter planPrinter, boolean verbose, Integer level)
+    {
+        PlanRepresentation plan = planPrinter.getPlanRepresentation();
```

Comment:
The method only needs the PlanRepresentation, so pass it directly instead of passing the PlanPrinter.